plugins/modules/edpm_nova_derive_pci_passthrough_devicespec.py

```diff
@@ -0,0 +1,649 @@
+        passthrough_tmp, status = get_passthrough_config(
+            user_config, pf, allocated_pci, False)
+        del_user_config = del_user_config or status
+        if passthrough_tmp:
+            derived_config.extend(passthrough_tmp)
+
+    """
+    If there is no config added from NIC Part nics, then skip parsing the
+    NON NIC Partitioned PFs.
+    """
+    if (derived_config or del_user_config):
+        for pf in non_nic_partition_pfs:
+            passthrough_tmp, status = get_passthrough_config(
+                user_config, pf, allocated_pci, True)
+            derived_config.extend(passthrough_tmp)
+
+    if derived_config:
+        return derived_config, False
+    else:
+        return derived_config, del_user_config
+
+
+def get_pf_name_from_phy_network(physical_network, phys_map=None):
+    # Read sriov phydev mapping
+    if phys_map and isinstance(phys_map, str):
+        # Need to check if it startswith(valid_spec)
+        phy = phys_map.strip()
+        map = re.split(',', phy)
+        for i in map:
+            net_name, nic_name = i.split(':')
+            if net_name == physical_network:
+                return nic_name
+    msg = f'physical_device_mappings specified {phys_map} is not in list'
+    raise InvalidConfigException(msg)
+    return None
+
+
+def generate_combined_configuration(user_configs, system_configs,
+                                    phys_dev_map=None):
+    """Derived configuration = user_config - system_configs
+
+    Identify the user_defined configuration that overlaps with the
+    NIC Partitioned VFs and remove those VFs from the derived configuration
+    In case of no overlap, the user defined configuration shall be used
+    as it is.
+    :param user_configs: device_spec list from nova conf
+    :param system_configs: Derived from sriov-mapping.yaml
+    :param phys_dev_map: Sriov physical device mannping
+    :return user_config_copy: Any non-nic partinioned cfg will be returned in this list
+    :return derived_config: All nic partinioned cfg will be returned after derivation in this list
+    """
+
+    user_config_copy = []
+    derived_config = []
+
+    allocated_pci = get_allocated_pci_addresses(system_configs)
+    nic_partition_pfs = get_sriov_nic_partition_pfs(system_configs)
+
+    for user_config in user_configs:
+        if 'address' in user_config and 'devname' in user_config:
+            msg = f"Both devname and address can't be present in {_PASSTHROUGH_DEVSPEC_KEY}"
+            raise InvalidConfigException(msg)
+
+        keys = ['address', 'product_id', 'devname']
+        if not any(k in user_config for k in keys):
+            # address or product_id or devname not present in user_config
+            pf = get_pf_name_from_phy_network(user_config['physical_network'],
+                                              phys_dev_map)
+            user_config['address'] = get_pci_addr_from_ifname(pf)
+
+        if 'devname' in user_config:
```

Comment:
we should be removing all usages fo devname.

its not something that customer should be using in there config.